### PR TITLE
Allow max_length 0 in dict_generic

### LINKED ISSUE #3: dict_generic has minimum length 1
First of all thank you for creating this library!

While using this library I noticed that `dict_generic` generates dictionaries with [a minimum length of 1](https://github.com/mooreryan/gleam_qcheck/blob/7ce8170570f54d8f32fcf6b4ff97add9b894a8af/src/qcheck/generator.gleam#L829). This is in contrast to [`set_generic`](https://github.com/mooreryan/gleam_qcheck/blob/7ce8170570f54d8f32fcf6b4ff97add9b894a8af/src/qcheck/generator.gleam#L811) which generates sets at a minimum size of zero.

I ran into this in a use-case where the maximum dictionary size I want to generate depends on another generator and can go down to zero, which triggers a failure due to [this assertion](https://github.com/mooreryan/gleam_qcheck/blob/7ce8170570f54d8f32fcf6b4ff97add9b894a8af/src/qcheck/generator.gleam#L335).

Was there any particular reason to choose a minimum size of 1 for dictionary generation? If not it may be better to allow a minimum size of zero.

### PR DESCRIPTION
**Pull request by aherrmann**
Closes #3

- **Test generation of empty dict**
- **Generate dict with min size 0**
- **Update birdie dict shrink test**

### CODE REVIEW

File: birdie_snapshots/dict_generators_shrink_on_size_then_on_elements_test.accepted [hunk 1]
```diff
@@ -4,55 +4,19 @@ title: dict_generators_shrink_on_size_then_on_elements__test
 file: ./test/qcheck/gen_dict_test.gleam
 test_name: dict_generators_shrink_on_size_then_on_elements__test
 ---
```
Comment (aherrmann) [inline birdie_snapshots/dict_generators_shrink_on_size_then_on_elements_test.accepted:6]: I'm not exactly sure if this matches the expected outcome. Please verify.

**Comment by mooreryan** [comment]
Thanks for the pull request.  If you could address the couple of comments, then I will accept it.

**Comment by aherrmann** [comment]
I ran `gleam format` to address the failure on CI.
I'll be happy to address any comments, however, I don't currently see any on GitHub.

**Comment by mooreryan** [comment]
> I'll be happy to address any comments, however, I don't currently see any on GitHub.

Sorry about that, I forgot to submit them.

File: test/qcheck/gen_dict_test.gleam [hunk 2]
```diff
@@ -53,10 +53,21 @@ pub fn dict_generators_shrink_on_size_then_on_elements__test() {
         value_generator: generator.int_uniform_inclusive(10, 12),
         max_length: 3,
       ),
-      seed.new(2),
+      seed.new(3),
```
[the remark this answers is not in the document]
Comment (aherrmann) [reply test/qcheck/gen_dict_test.gleam]: No problem, I've set it to 12.

**Comment by aherrmann** [comment]
Thank you! I've updated the PR.

**Comment by mooreryan** [comment]
The CI checks are failing, but it is due to the formatting.  This workflow had been on an older version of Gleam and so had slightly different formatting, and I figured that you may have been using the latest Gleam.  I have updated the repository and workflow to use the formatting from Gleam version 1.2.1.  After re-running the workflow, I see the formatting checks are still failing.  So, if you can address that, everything else looks fine.

**Comment by mooreryan** [comment]
Thanks again for this!!

**Pull request merged**